backend/src/index.ts

```diff
@@ -0,0 +1,151 @@
+import express, { Request, Response } from 'express';
+import cors from 'cors';
+import { prisma } from './database'
+import { errorResponseBuilder, generateRandomPassword, isValidUser, parseUserForResponse } from './utils';
+
+const app = express();
+app.use(express.json());
+app.use(cors())
+
+// Create new user
+app.post('/users/new', async (req: Request, res: Response) => {
+  const errorBuilder = errorResponseBuilder(res);
+
+  try {
+    const userData = req.body;
+
+    // Validate input
+    if(!isValidUser(req.body)){
+      return errorBuilder.validationError()
+    }
+    
+    // Check if the user exists or not
+    const existingUserByEmail = await prisma.user.findFirst({ where: { email: req.body.email }});
```

Comment:
Maybe you could put prisma-related code to a directory called model or db and put the code inside it

Also maybe we can use .findUnique instead of .findFirst because we explicitly emphasize that the email is unique

Reply:
Good idea, I'll outsource the db queries & mutations to `database.ts`.
The comment about using `findUnique` is also on point, thanks!